executor/index_lookup_join.go

```diff
@@ -355,21 +376,30 @@ func (ow *outerWorker) increaseBatchSize() {
 }
 
 func (iw *innerWorker) run(ctx context.Context, wg *sync.WaitGroup) {
-	defer wg.Done()
-	for {
+	var task *lookUpJoinTask
+	defer func() {
+		if r := recover(); r != nil {
+			// "task != nil" is guaranteed when panic happened.
+			task.doneCh <- errors.Errorf("%v", r)
+		}
+		wg.Done()
+	}()
+
+	for ok := true; ok; {
 		select {
-		case task, ok := <-iw.taskCh:
+		case task, ok = <-iw.taskCh:
 			if !ok {
 				return
 			}
-			err := iw.handleTask(ctx, task)
-			task.doneCh <- errors.Trace(err)
-			if err != nil {
-				return
-			}
 		case <-ctx.Done():
 			return
 		}
+
+		err := iw.handleTask(ctx, task)
+		task.doneCh <- errors.Trace(err)
```

Comment:
I wonder is it really necessary to send `nil` to `taskk.doneCh` ?

Reply:
We can close `task.doneCh` when `err == nil` actually, but send a `nil` is simpler.